### PR TITLE
Detail Field on tasks

### PR DESCRIPTION
**Pull request by Tarrke**
This should fixe #45 
* updated UI
* updated data storage
* updated tests
* updated locales

### CODE REVIEW

**Comment by Tarrke** [comment]
Ho crap this had the #60 fixes within... If this is a problem I'll solve it once the other PR is merged. The duplicated commits will naturaly disaper then.

**Comment by JonasJurczok** [comment]
Cool :)
That's no problem at all. Git will solve this :D

File: Tasksets/sets.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Sets of tasks
+
+## A simple task set
+
+This set is just a simple set that comports 3 tasks. It is mainly an example of the task export system.
+
+```lua
+W3sidGFzayI6IkEgc2ltcGxlIHRhc2siLCJ0aXRsZSI6IlRhc2sgMSJ9LHsidGFzayI6IkFuIG90aGVyIHNpbXBsZSB0YXNrIiwidGl0bGUiOiJUYXNrIDIifSx7InRhc2siOiJZZXQgYW4gb3RoZXIgdmVyeSBzaW1wbGUgdGFzayIsInRpdGxlIjoiVGFzayAzIn1d
+```
```
Comment (JonasJurczok) [inline Tasksets/sets.md]: The string is different from the other PR :D

File: spec/UI_feature.lua [hunk 2]
```diff
@@ -2,8 +2,8 @@ local task_num = 1
 function add_task(text)
     text = text or 'this is a task ' .. task_num
     task_num = task_num + 1
-
-    local task = todo.create_task(text)
+    local task_template = { ["task"] = text, ["title"] = "Title", ["assignee"] = "def" }
+    local task = todo.create_task(task_template, "Jonas")
```
Comment (JonasJurczok) [inline spec/UI_feature.lua]: :D

File: spec/UI_spec.lua [hunk 3]
```diff
@@ -11,15 +11,15 @@ describe("UI tests", function()
 	faketorio.initialize_world_busted()
     faketorio.add_default_setting("todolist-show-button", true)
     faketorio.add_default_setting("todolist-show-log", false)
-	todo.mod_init()	
+	todo.mod_init()
```
Comment (JonasJurczok) [inline spec/UI_spec.lua]: This reminds me: I have to rework that file into faketorio tests -.-

File: src/todo/todo.lua [hunk 4]
```diff
@@ -145,38 +152,45 @@ function todo.import_tasks(dialog)
...
     frame.destroy()
 end
 
 function todo.get_task_from_add_frame(frame)
-    local taskText = frame.todo_add_task_table.children[2].text
+    local taskText = frame.todo_add_task_table["todo_new_task_textbox"].text
+    local taskTitle = frame.todo_add_task_table["todo_new_task_title"].text
 
-    local assignees = frame.todo_add_task_table.children[4]
+    local assignees = frame.todo_add_task_table["todo_add_assignee_drop_down"]
```
Comment (JonasJurczok) [inline src/todo/todo.lua]: nice!

File: Tasksets/sets.md (hunk 1 above)
Comment (Tarrke) [reply Tasksets/sets.md]: It is different as I had to now save the title and the task. More info to stock.

File: src/todo/todo.lua (hunk 4 above)
Comment (Tarrke) [reply src/todo/todo.lua]: I think we'd better avoid the children[id] everywhere. If we add some things before id 4 this won't work anymore.

**Comment by Tarrke** [comment]
Yes I'll wait for the other PR to be merged then rebase the whole branch, should be a little simpler to do then.

**Comment by JonasJurczok** [comment]
@Tarrke I rebased the branch on the current master.
I'll try to verify that the changes are still as they should be and then will try to complete the PR. Feel free to add comments if you see things missing.

**Comment by JonasJurczok** [comment]
Missing Test:
Create task with just title and no details.
check if editing works
check if exporting/importing works

**Comment by JonasJurczok** [comment]
No problem. I actually really enjoyed that :)

**Pull request merged**